device_e2e/aio/retry_async.py

```diff
@@ -0,0 +1,179 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+import asyncio
+import logging
+import random
+import threading
+import time
+from azure.iot.device.exceptions import (
+    ConnectionFailedError,
+    ConnectionDroppedError,
+    OperationCancelled,
+    NoConnectionError,
+)
+
+logger = logging.getLogger(__name__)
+logger.setLevel(level=logging.INFO)
+
+# --------------------------------------
+# Parameters for our backoff and jitter
+# --------------------------------------
+
+# Retry immediately after failure, or wait unti after first delay?
+IMMEDIATE_FIRST_RETRY = True
+
+# Seconds to sleep for first sleep period. The exponential backoff will use
+# 2x this number for the second sleep period, then 4x this number for the third
+# period, then 8x and so on.
+INITIAL_DELAY = 5
+
+# Largest number of seconds to sleep between retries (before applying jitter)
+MAXIMUM_DELAY = 60
+
+# Number of seconds before an operation is considered "failed". This period starts before
+# the first attempt and includes the elapsed time waiting between any failed attempts.
+FAILURE_TIMEOUT = 5 * 60
+
+# Jitter-up factor. The time, after jitter is applied, can be up this percentage larger than the
+# pre-jittered time.
+JITTER_UP_FACTOR = 0.25
```

Comment:
is there any reason for the choice of these 2 values to be different ?

Reply:
@olivakar - it's entirely arbitrary. All of our other implementations have 2 different numbers, so I did too.